src/main/java/org/broadinstitute/hellbender/tools/walkers/genotyper/AlleleLikelihoodMatrixMapper.java

```diff
@@ -12,27 +13,31 @@
 /**
  * Creates {@link org.broadinstitute.hellbender.utils.genotyper.LikelihoodMatrix} mappers to be used when working with a subset of the original alleles.
  */
-@FunctionalInterface
-public interface AlleleLikelihoodMatrixMapper<A extends Allele> extends UnaryOperator<LikelihoodMatrix<A>>{
+public class AlleleLikelihoodMatrixMapper<A extends Allele> {
 
+    private final AlleleListPermutation<A> permutation;
     /**
-     * Instantiates a new mapper given an allele-list permutation.
+     * Constructs a new mapper given an allele-list permutation.
      * @param permutation the requested permutation.
-     * @param <A> the allele type.
      *
      * @throws IllegalArgumentException if {@code permutation} is {@code null}.
      *
      * @return never {@code null}.
      */
-    public static <A extends Allele> AlleleLikelihoodMatrixMapper<A> newInstance(final AlleleListPermutation<A> permutation) {
-        Utils.nonNull(permutation, "the permutation must not be null");
+    public AlleleLikelihoodMatrixMapper(final AlleleListPermutation<A> permutation) {
+        this.permutation = Utils.nonNull(permutation);
+    }
+
+    @SuppressWarnings({"unchecked", "rawtypes"})
+    public <EVIDENCE> LikelihoodMatrix<EVIDENCE, A> mapAlleles(final LikelihoodMatrix<EVIDENCE, A> original) {
```

Comment:
If this class represented the posterior probability Prob(allele | reads) I would agree, but since it is used for likelihoods Prob(read | allele) I would argue for keeping the `<EVIDENCE, A>` order.